test/new-e2e/tests/agent-metric-logs/kindfilelogging/kind.go

```diff
@@ -0,0 +1,216 @@
+	// We ALWAYS need to make a deep copy of `params`, as the provisioner can be called multiple times.
+	// and it's easy to forget about it, leading to issues that are hard to debug.
+	params := newProvisionerParams()
+	_ = optional.ApplyOptions(params, opts)
+
+	provisioner := e2e.NewTypedPulumiProvisioner(provisionerBaseID+params.name, func(ctx *pulumi.Context, env *environments.Kubernetes) error {
+		// We ALWAYS need to make a deep copy of `params`, as the provisioner can be called multiple times.
+		// and it's easy to forget about it, leading to issues that are hard to debug.
+		params := newProvisionerParams()
+		_ = optional.ApplyOptions(params, opts)
+
+		return KindRunFunc(ctx, env, params)
+	}, params.extraConfigParams)
+
+	return provisioner
+}
+
+// KindRunFunc is the Pulumi run function that runs the provisioner
+func KindRunFunc(ctx *pulumi.Context, env *environments.Kubernetes, params *ProvisionerParams) error {
+	awsEnv, err := aws.NewEnvironment(ctx)
+	if err != nil {
+		return fmt.Errorf("aws.NewEnvironment: %w", err)
+	}
+
+	host, err := ec2.NewVM(awsEnv, params.name, params.vmOptions...)
+	if err != nil {
+		return fmt.Errorf("ec2.NewVM: %w", err)
+	}
+
+	installEcrCredsHelperCmd, err := ec2.InstallECRCredentialsHelper(awsEnv, host)
+	if err != nil {
+		return fmt.Errorf("ec2.InstallECRCredentialsHelper %w", err)
+	}
+
+	kindCluster, err := kubeComp.NewKindCluster(*awsEnv.CommonEnvironment, host, awsEnv.CommonNamer.ResourceName("kind"), params.name, awsEnv.KubernetesVersion(), utils.PulumiDependsOn(installEcrCredsHelperCmd))
+	if err != nil {
+		return fmt.Errorf("kubeComp.NewKindCluster: %w", err)
+	}
+	err = kindCluster.Export(ctx, &env.KubernetesCluster.ClusterOutput)
+	if err != nil {
+		return fmt.Errorf("kindCluster.Export: %w", err)
+	}
+
+	kubeProvider, err := kubernetes.NewProvider(ctx, awsEnv.Namer.ResourceName("k8s-provider"), &kubernetes.ProviderArgs{
+		EnableServerSideApply: pulumi.Bool(true),
+		Kubeconfig:            kindCluster.KubeConfig,
+	})
+	if err != nil {
+		return fmt.Errorf("kubernetes.NewProvider: %w", err)
+	}
+
+	if params.fakeintakeOptions != nil {
+		fakeintakeOpts := []fakeintake.Option{fakeintake.WithLoadBalancer()}
+		params.fakeintakeOptions = append(fakeintakeOpts, params.fakeintakeOptions...)
+		fakeIntake, err := fakeintake.NewECSFargateInstance(awsEnv, params.name, params.fakeintakeOptions...)
+		if err != nil {
+			return fmt.Errorf("fakeintake.NewECSFargateInstance: %w", err)
+		}
+		err = fakeIntake.Export(ctx, &env.FakeIntake.FakeintakeOutput)
+		if err != nil {
+			return fmt.Errorf("fakeIntake.Export: %w", err)
+		}
+
+		if params.agentOptions != nil {
+			newOpts := []kubernetesagentparams.Option{kubernetesagentparams.WithFakeintake(fakeIntake)}
+			params.agentOptions = append(newOpts, params.agentOptions...)
+		}
+	} else {
+		env.FakeIntake = nil
+	}
+
+	if params.agentOptions != nil {
+		kindClusterName := ctx.Stack()
+		helmValues := fmt.Sprintf(`
+datadog:
```

Comment:
Just wondering if we should create a folder for different files of helm values and parse it as string here? Since we potentially will have multiple configuration

Reply:
That's a good idea, I can definitely do that in a later change.